STACpopulator/extensions/cordex6.py

```diff
@@ -0,0 +1,95 @@
+from __future__ import annotations
+
+from pathlib import Path
+from pydantic import BaseModel, Field, FilePath
+from datetime import datetime
+
+from importlib import reload
+import STACpopulator.extensions.base
+reload(STACpopulator.extensions.base)
+from STACpopulator.extensions.base import THREDDSCatalogDataModel, DataModel
+
+
+# This is generated using datamodel-codegen + manual edits
+class CordexCmip6(DataModel):
```

Comment:
Since the model is generated from the schema, why is the `@model_validator` needed to load and validate the JSON schema?

I'm not seeing the subtlety from static code analysis.

Reply:
It's slimmed down version of the schema without the actual CV validation. The schema includes enums with the CVs, while the pydantic.DataModel does not. 

This is a question I struggled with. I felt it didn't make a lot of sense to duplicate the jsonschema validation in pydantic. On the other hand, relying only on the schema and not even seeing the attributes in the code felt obscure and not admin friendly. So I thought it would be useful to have a `pydantic DataModel` layer where you can add attributes to the data model, and exclude some that are in the schema but you don't want in the STAC item.

Reply:
I wonder if that is an issue with datamodel-codegen, or an option to provide? 
Normally, the enums should be possible using `Literal` type with pydantic. 

I think it makes sense to have the `DataModel` auto-generated from schema to provide the attributes. It's easier to manipulate by users used to Python but not so much JSON schema.

Reply:
Yes, it's definitely possible. I just didn't include the Literals in the python code.